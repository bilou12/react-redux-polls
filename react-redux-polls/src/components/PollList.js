import { connect } from "react-redux";
import Poll from "./Poll.js";
import { userHasAlreadyVoted } from "../utils/helper.js";

const PollList = ({ questionValues, authedUser }) => {
  let questionsDone = questionValues.filter((question) => {
    return userHasAlreadyVoted(question, authedUser);
  });

  let questionsTodo = questionValues.filter((question) => {
    return !userHasAlreadyVoted(question, authedUser);
  });

  return (
    <div>
      <h3>Poll List</h3>
      <h4>Todo</h4>
      <ul className="question-list">
        {questionsTodo.map((question) => {
          // return <li key={question.id}> {question.id}</li>;
          return <Poll question={question} key={question.id}></Poll>;
        })}
      </ul>

      <h4>Done</h4>
      <ul className="question-list">
        {questionsDone.map((question) => {
          // return <li key={question.id}> {question.id}</li>;
          return <Poll question={question} key={question.id}></Poll>;
        })}
      </ul>
    </div>
  );
};

const mapStateToProps = ({ questions, authedUser }) => ({
  questionValues: Object.values(questions),
  authedUser,
});

export default connect(mapStateToProps)(PollList);
