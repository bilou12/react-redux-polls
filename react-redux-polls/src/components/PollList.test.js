import React from "react";
import PollList from "./PollList";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import configureMockStore from "redux-mock-store";

const mockStore = configureMockStore();
const store = mockStore({
  questions: {
    question1: {
      id: "question1",
      author: "user1",
      optionOne: {
        votes: ["user2"],
        text: "Option One",
      },
      optionTwo: {
        votes: ["user1"],
        text: "Option Two",
      },
      timestamp: 123456789,
    },
    question2: {
      id: "question2",
      author: "user2",
      optionOne: {
        votes: ["user2", "user3"],
        text: "Option One",
      },
      optionTwo: {
        votes: ["user1"],
        text: "Option Two",
      },
      timestamp: 987654321,
    },
  },
  authedUser: "user1",
});

describe("PollList component", () => {
  it("should render correctly with questions to do", () => {
    const { container } = render(
      <Provider store={store}>
        <PollList />
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
