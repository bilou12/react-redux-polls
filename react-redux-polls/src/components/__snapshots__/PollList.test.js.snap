// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollList component should render correctly with questions to do 1`] = `
<div>
  <div>
    <h3>
      Poll List
    </h3>
    <select>
      <option
        value="todo"
      >
        To do
      </option>
      <option
        value="done"
      >
        Done
      </option>
    </select>
    <h4>
      todo
    </h4>
    <ul
      class="question-list"
    />
  </div>
</div>
`;
